packages/nimble-components/src/tree-view/index.ts

```diff
@@ -29,6 +30,14 @@ export class TreeView extends FoundationTreeView {
     @attr({ attribute: 'selection-mode' })
     public selectionMode: TreeViewSelectionMode = TreeViewSelectionMode.all;
 
+    private readonly baseHandleSelectedChange: (event: Event) => void;
+
+    public constructor() {
+        super();
+        this.baseHandleSelectedChange = this.handleSelectedChange;
+        this.handleSelectedChange = event => this.handleSelectedChangeOverride(event);
```

Comment:
I wanted to override the TreeView's existing handler for the `selected-changed` event, but I had some trouble. The handler is defined as a property (rather than a function), and I could not call the `super.handleSelectedChanged` from my override. I had to cache the base implementation, then call the cached function from my override. If there's some cleaner way to do this, let me know. Apparently the property is not on the object prototype.

Reply:
Would it work to just register a new event listener for the event and rely on [standard event handling mechanisms for ordering handlers and preventing propagation if needed](https://stackoverflow.com/questions/16273635/how-do-multiple-addeventlistener-work-in-javascript)?

Reply:
Yes, I think that would work, too. I had originally considered that, but then got occupied with trying to make overriding work. I guess just adding a second event listener is cleaner at this point.

Reply:
Yeah, my suspicion is that if we register the handler in the constructor, that should get registered _before_ the handler does on the base class, which is hooked up in the template code. That said, it seems sketchy to rely on that ordering.

Reply:
Oh right, since we're always calling the base implementation.